mlir/lib/Dialect/MIOpen/CoordTransformBuilder.cpp

```diff
@@ -112,6 +112,17 @@ AffineMapAttr assembleMapFor(Builder &b, ArrayRef<TransformAttr> transforms,
       assert(upperDims.size() == 1 && lowerDims.size() == 0 &&
              "Invalid AddDim");
       // dimension is ignored, do nothing
+    } else if (type == TransformType::Broadcast) {
+      // Compute lower dimension strides.
+      for (auto pair : llvm::enumerate(lowerBounds)) {
```

Comment:
Not like this, please

Proposed alternate design: broadcast takes a set of upper dimensions and a set of lower dimensions, and its parameters are the sizes of the lower dimensions (yes, it's duplicative, it's future-proofing for dynamic shapes)

That is, `broadcast{2, 1} ["g", "h"] at [0, 1] -> ["g", "h"] at [0, 1]` produces the map `(d0, d1) -> (d0 % 2, d1 % 1)`, aka `(d0, d1) -> (d0 % 2, 0)`. The upper size is specified separately.

The individual transforms doing this sort of global reasoning over the entire map makes the setup much less clean.

Reply:
yes, I think I understand now. lowerBounds/upperBounds are global? and the lowerDims/upperDims are specific to the transform?

Reply:
Yep, lowerBounds is a global property that shouldn't come up in the affine map construction, and should be instead handled by setting parameters on the transform.

This is partly because, from where I'm standing, it might make sense to remove upperBounds and lowerBounds entirely, especially now that we have `transforming_for`